santosfamilyfoundation/RuinaMain
# [RUIN-269] Reduce effects of question loading lagging and add loading spinner on question form

# Description

Previously, the questions on the question form took an extremely long time to load, which was frustrating. This change attempts to reduce some of this lagging by reducing the repetitive computation needed to load questions. However, the change in lag was not significant after this, so additionally, I have added a loading spinner to the question form to visually communicate to the user that the questions are loading. While still frustrating, the loading spinner makes the wait less confusing. Regardless, in the future, we should try to address the underlying lag issue.

# Testing
1. Open the app and create a new report.
2. Add one vehicle and one non-motorist.
3. Click on each form section in the Home screen. Each section should initially render with a loading spinner, and then with the questions.
